sklearn/gaussian_process/_gpr.py

```diff
@@ -343,15 +352,20 @@ def predict(self, X, return_std=False, return_cov=False):
             else:
                 return y_mean
         else:  # Predict based on GP posterior
+            # Alg 2.1, page 19, line 4 -> f*_bar = K(X_test, X_train) . alpha
             K_trans = self.kernel_(X, self.X_train_)
-            y_mean = K_trans.dot(self.alpha_)  # Line 4 (y_mean = f_star)
+            y_mean = K_trans.dot(self.alpha_)
 
             # undo normalisation
             y_mean = self._y_train_std * y_mean + self._y_train_mean
 
             if return_cov:
-                v = cho_solve((self.L_, True), K_trans.T)  # Line 5
-                y_cov = self.kernel_(X) - K_trans.dot(v)  # Line 6
+                # Alg 2.1, page 19, line 5 -> v = L \ K(X_test, X_train)^T
+                v = cho_solve(
+                    (self.L_, self._L_lower_flag), K_trans.T
+                )
+                # Alg 2.1, page 19, line 6 -> K(X_test, X_test) - v . v^T
+                y_cov = self.kernel_(X) - K_trans.dot(v)
```

Comment:
Here the computations from the implementation and the reference are different but the result equivalent.

As `v` from the code and the one from the comment differ, probably we could use another variable name here and indicate that we solve it differently (note that using `cho_solve` should solve two triangular systems)?
```suggestion
                # Alg 2.1, page 19, line 5 -> v = L \ K(X_test, X_train)^T
                z = cho_solve(
                    (self.L_, self._L_lower_flag), K_trans.T
                )
                # Alg 2.1, page 19, line 6 -> K(X_test, X_test) - v . v^T
                y_cov = self.kernel_(X) - K_trans.dot(z)
```

or have computations match the algorithm (this should solve only one triangular system):

```suggestion
                # Alg 2.1, page 19, line 5 -> v = L \ K(X_test, X_train)^T
                v = solve_triangular(self.L_, K_trans.T, lower=self._L_lower_flag)
                # Alg 2.1, page 19, line 6 -> K(X_test, X_test) - v . v^T
                y_cov = self.kernel_(X) - v.T.dot(v)
```

Reply:
It seems that reformulating with a single triangular system to solve is better.
I think that we can delay this PR after including the bug fix #19939 such that the covariance and std. dev. computation, annotations are consistent.